vllm/v1/sample/rejection_sampler.py

```diff
@@ -191,3 +266,56 @@ def _create_greedy_token_probs(
                          src=valid_mask.unsqueeze(-1).float())
 
     return token_probs
+
+
+def _create_uniform_samples(seeded_seqs: Optional[Dict[int, torch.Generator]],
+                            batch_size: int, k: int,
+                            device: torch.device) -> torch.Tensor:
+    """
+        Generates a batch of uniform random samples, with optional seeding 
+        for specific sequences.
+
+        This method creates a tensor of shape `(batch_size, k)` filled 
+        with uniform random values in the range [0, 1). If `seeded_seqs` 
+        is provided, the sequences corresponding to specific indices 
+        will be generated using the provided `torch.Generator` for 
+        reproducibility. The other sequences will be generated without 
+        a seed.
+
+        Args:
+            seeded_seqs : Optional[Dict[int, torch.Generator]]
+                A dictionary mapping indices in the batch to 
+                `torch.Generator` objects. If `None`, all samples are 
+                generated without a seed.
+            batch_size : int
+                The number of sequences to generate.
+            k : int
+                The number of random samples per sequence.
+            device : torch.device
+                The device on which to allocate the tensor.
+
+        Returns:
+            uniform_rand : torch.Tensor
+                A tensor of shape `(batch_size, k)` containing uniform 
+                random values in the range [0, 1).
+        """
+    if not seeded_seqs:
+        return torch.rand(batch_size, k, device=device)
+
+    uniform_rand = torch.empty(batch_size, k, device=device)
+
+    non_seeded_indices = []
+    for idx in range(batch_size):
+        generator = seeded_seqs.get(idx)
+        if generator is None:
+            non_seeded_indices.append(idx)
+        else:
+            uniform_rand[idx, :] = torch.rand(1,
+                                              k,
+                                              dtype=torch.float,
+                                              device=device,
+                                              generator=generator)
+    if non_seeded_indices:
+        uniform_rand[non_seeded_indices, :] = torch.rand(
+            len(non_seeded_indices), k, dtype=torch.float, device=device)
```

Comment:
this looks fairly inefficient. Why don't we assume no generator is given, create a random tensor, and then loop through *the generators* instead of the whole batch?

smt like

```suggestion
   uniform_rand = torch.rand(batch_size, k, device=device)

    # Apply seeded generators only where needed
    if seeded_seqs:
        for idx, generator in seeded_seqs.items():
            uniform_rand[idx, :] = torch.rand(1, k, device=device, generator=generator)

    return uniform_rand

```

Reply:
Got it, thanks!